commit 5a5436b98d5cd2714feaaa579cec49dd7f7057bb
Author: Paolo Valente <paolo.valente@linaro.org>
Date:   Mon Jan 25 20:02:47 2021 +0100

    block, bfq: save also injection state on queue merging
    
    To prevent injection information from being lost on bfq_queue merging,
    also the amount of service that a bfq_queue receives must be saved and
    restored when the bfq_queue is merged and split, respectively.
    
    Tested-by: Jan Kara <jack@suse.cz>
    Signed-off-by: Paolo Valente <paolo.valente@linaro.org>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/block/bfq-iosched.h b/block/bfq-iosched.h
index d15299d59f89..3f350fa3c5fd 100644
--- a/block/bfq-iosched.h
+++ b/block/bfq-iosched.h
@@ -444,6 +444,11 @@ struct bfq_io_cq {
 	unsigned long saved_wr_start_at_switch_to_srt;
 	unsigned int saved_wr_cur_max_time;
 	struct bfq_ttime saved_ttime;
+
+	/* Save also injection state */
+	u64 saved_last_serv_time_ns;
+	unsigned int saved_inject_limit;
+	unsigned long saved_decrease_time_jif;
 };
 
 /**